packages/react/src/primitives/types/textField.ts

```diff
@@ -36,4 +36,6 @@ export interface TextFieldOptions extends FieldProps, FlexContainerStyleProps {
   type?: InputProps['type'];
 }
 
-export interface TextFieldProps extends TextFieldOptions, InputProps {}
+export interface TextFieldProps
+  extends TextFieldOptions,
+    Omit<InputProps, 'checked'> {}
```

Comment:
Not entirely sure we can remove this without considering it a breaking change :(

Reply:
Technically yes... but maybe not in the spirit of the law? This was a bug that we were exposing this. But realistically it might NOT be a breaking change because.... the interface for an HTMLInputElement coming from React types includes the `checked` attribute and because we are extending from the element's type, this won't actually throw any TS errors. 

<img width="799" alt="CleanShot 2023-01-10 at 09 47 54@2x" src="https://user-images.githubusercontent.com/321279/211624889-210c2cb6-4147-4073-b9b0-ee8f4a8241ef.png">

<img width="1232" alt="CleanShot 2023-01-10 at 09 48 29@2x" src="https://user-images.githubusercontent.com/321279/211625056-f48162fd-816c-41c6-a04f-516cdcc75ef1.png">

<img width="1229" alt="CleanShot 2023-01-10 at 09 45 57@2x" src="https://user-images.githubusercontent.com/321279/211624829-4fdb9658-4c9d-4cff-b50e-682265f3c027.png">

So maybe this change (or a similar change, see @ioanabrooks's comment) is what we want. We remove it from our types and documentation, but won't break anyone's build.

Reply:
It should be fine because `TextField` never have to use `checked` and can't use `checked`

Reply:
Are we absolutely certain that there is no customer impact? Is there potential that a user erroneously added the `checked` prop to one of the impacted UI components and that they will see type errors when it is removed?